### PR TITLE
core: allow passing None for optional OpResult

### PR DESCRIPTION
**Pull request by superlopuh**

### CODE REVIEW

File: xdsl/irdl.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def __init__(
         self: IRDLOperation,
         operands: Sequence[SSAValue | Operation | Sequence[SSAValue | Operation] | None]
         | None = None,
-        result_types: Sequence[Attribute | Sequence[Attribute]] | None = None,
+        result_types: Sequence[Attribute | Sequence[Attribute] | None] | None = None,
```
Comment (superlopuh) [inline xdsl/irdl.py:370]: logic was already implemented

**Review by math-fehr — APPROVED** [review]
Nice!

**Comment by math-fehr** [comment]
Can you rebase it, so I can merge it?
(I think the black PR was merge after this PR?)

**Comment by superlopuh** [comment]
It's already rebased :)

**Pull request merged**